src/backend/distributed/commands/comment.c

```diff
@@ -0,0 +1,137 @@
+/*-------------------------------------------------------------------------
+ *
+ * comment.c
+ *    Commands to interact with the comments for all database
+ *    object types.
+ *
+ * Copyright (c) Citus Data, Inc.
+ *
+ *-------------------------------------------------------------------------
+ */
+
+#include "postgres.h"
+
+#include "access/genam.h"
+#include "access/htup_details.h"
+#include "access/table.h"
+#include "catalog/pg_shdescription.h"
+#include "nodes/parsenodes.h"
+#include "utils/builtins.h"
+#include "utils/fmgroids.h"
+#include "utils/rel.h"
+
+#include "distributed/comment.h"
+
+static char * GetCommentForObject(Oid classOid, Oid objectOid);
+
+
+List *
+GetCommentPropagationCommands(Oid classOid, Oid objOoid, char *objectName, ObjectType
+							  objectType)
+{
+	List *commands = NIL;
+
+	StringInfo commentStmt = makeStringInfo();
+
+	/* Get the comment for the database */
+	char *comment = GetCommentForObject(classOid, objOoid);
+	char const *commentObjectType = ObjectTypeNames[objectType];
+
+	/* Create the SQL command to propagate the comment to other nodes */
+	if (comment != NULL)
+	{
+		appendStringInfo(commentStmt, "COMMENT ON %s %s IS %s;", commentObjectType,
+						 quote_identifier(objectName),
+						 quote_literal_cstr(comment));
+	}
+
+
+	/* Add the command to the list */
+	if (commentStmt->len > 0)
+	{
+		commands = list_make1(commentStmt->data);
+	}
+
+	return commands;
+}
+
+
+static char *
+GetCommentForObject(Oid classOid, Oid objectOid)
+{
+	HeapTuple tuple;
+	char *comment = NULL;
+
+	/* Open pg_shdescription catalog */
+	Relation shdescRelation = table_open(SharedDescriptionRelationId, AccessShareLock);
+
+	/* Scan the table */
+	ScanKeyData scanKey[2];
+
+	ScanKeyInit(&scanKey[0],
+				Anum_pg_shdescription_objoid,
+				BTEqualStrategyNumber, F_OIDEQ,
+				ObjectIdGetDatum(objectOid));
+	ScanKeyInit(&scanKey[1],
+				Anum_pg_shdescription_classoid,
+				BTEqualStrategyNumber, F_OIDEQ,
+				ObjectIdGetDatum(classOid));
+	bool indexOk = true;
+	int scanKeyCount = 2;
+	SysScanDesc scan = systable_beginscan(shdescRelation, SharedDescriptionObjIndexId,
+										  indexOk, NULL, scanKeyCount,
+										  scanKey);
+	while ((tuple = systable_getnext(scan)) != NULL)
+	{
+		Form_pg_shdescription shdesc = (Form_pg_shdescription) GETSTRUCT(tuple);
+
+		bool isNull = false;
+
+		TupleDesc tupdesc = RelationGetDescr(shdescRelation);
+
+		Datum descDatum = heap_getattr(tuple, Anum_pg_shdescription_description, tupdesc,
+									   &isNull);
+
+		/* Check if the objoid matches the databaseOid */
+		if (shdesc->objoid == objectOid && shdesc->classoid == classOid)
+		{
```

Comment:
This is now not needed anymore. since the ScanKeyInit calls already take care of this.